query-graphs-js/src/querygraph.ts

```diff
@@ -61,6 +61,13 @@ export function isFederatedGraphRootType(type: NamedType) {
  */
 export class Vertex {
   hasReachableCrossSubgraphEdges: boolean = false;
+  // @provides works by creating duplicates of the vertex/type involved in the provides and adding the provided
+  // edges only to those copy. This means that with @provides, you can have more than one vertex per-type-and-subgraph
+  // in a query graph. Which is fined, but this `provideId` allows to distinguish if a vertex was created as part of
+  // this @provides duplication or not. The value of this field has no other meaning than to be unique per-@provide,
+  // and so all the vertex copied for a given @provides application will have the same `provideId`. Overall, this
+  // mostly exists for debugging visualisation.
+  provideId: number | undefined;
```

Comment:
Pretty much all the change in this file are just to populate this, because I noticed there wasn't a simple way to display things correctly in mermaid without that. I think it's kind of convenient to have for debugging in general, and it should be fairly obvious that this addition doesn't break anything.